### PR TITLE
Change requre: in config to be relative to the .rubocop.yml file itself

### LINKED ISSUE #2865: require: in rubocop.yml is relative to pwd not the file
When I do

```
require:
  ./foo/bar.rb
```

It changes which `bar.rb` I'm using based on which dir I'm running `rubocop` in. Worse, this breaks editor integrations because some of them have the `pwd` in `/tmp` somewhere with only your edited files.

### PR DESCRIPTION
**Pull request by ptarjan**
Fixes #2865

### CODE REVIEW

**Comment by rrosenblum** [comment]
Is it possible to add tests for this?

**Comment by ptarjan** [comment]
@rrosenblum updated with a test

**Comment by rrosenblum** [comment]
Thanks for adding tests. The change looks good to me.

File: lib/rubocop/config_loader.rb [hunk 1]
```diff
@@ -35,7 +35,10 @@ def load_file(path)
         resolve_inheritance_from_gems(hash, hash.delete('inherit_gem'))
         resolve_inheritance(path, hash)
 
-        Array(hash.delete('require')).each { |r| require(r) }
+        base_dir = File.dirname(path)
```
Comment (bbatsov) [inline lib/rubocop/config_loader.rb]: Maybe `config_dir` is a better name for this?

File: spec/rubocop/config_loader_spec.rb [hunk 2]
```diff
@@ -505,5 +505,13 @@
       described_class.configuration_from_file(config_path)
       expect(defined?(MyClass)).to be_truthy
     end
+
+    it 'uses paths relative to the yml, not cwd' do
```
Comment (bbatsov) [inline spec/rubocop/config_loader_spec.rb]: yml -> `.rubocop.yml`

**Comment by bbatsov** [comment]
Apart my small remark, the PR has to be rebased on top of the current `master`.

**Comment by ptarjan** [comment]
@bbatsov all your changes are applied and I rebased

**Comment by maciejmajewski** [comment]
It looks like this change introduced a lot of backwards incompatibility.

Could you tell how to include other GEMs in configuration (like `rubocop-rspec`) ?

Also the README "Loading Extensions" section [1] is no longer valid, as requiring 'rubocop-extension' will require './rubocop-extension' from now.

[1] https://github.com/bbatsov/rubocop#loading-extensions

**Comment by ptarjan** [comment]
@maciejmajewski sorry about that. What do you think about https://github.com/bbatsov/rubocop/pull/2936 ?

**Comment by maciejmajewski** [comment]
@ptarjan LGTM

**Pull request merged**